src/snowflake/snowpark/session.py

```diff
@@ -277,7 +278,13 @@ def __init__(self, conn: ServerConnection) -> None:
         self._file = FileOperation(self)
 
         self._analyzer = Analyzer(self)
-        self._sql_simplifier_enabled: bool = False
+        self._sql_simplifier_enabled: bool = (
```

Comment:
Since this is the second usage of the same pattern like `_use_scoped_temp_objects`, I think we should extract this into a helper/util function to avoid code duplication.

Reply:
Added an internal function `_get_client_side_session_parameter` on class Session.